Source/WebInspectorUI/UserInterface/Workers/Formatter/JSFormatter.js

```diff
@@ -161,7 +161,7 @@ JSFormatter = class JSFormatter
 
     _before(node)
     {
-        if (!node.parent)
+        if (!node.parent || !node.range)
```

Comment:
NIT: should we move these checks to `ESTreeWalker` instead so we dont have to have them in every `_before`/`_after`?

Reply:
We could, although that feels like it causes this implementation detail of JSFormatter to leak into ESTreeWalker. This is a case where we are making a decision based on what properties the node has what to do, which includes what to do if a range is not present.